src/hmem_cuda.c

```diff
@@ -760,6 +760,19 @@ bool cuda_is_gdrcopy_enabled(void)
 	return hmem_cuda_use_gdrcopy;
 }
 
+#if CUDA_VERSION >= 11030
+int cuda_flush_rdma_operations(void)
+{
+	return cudaDeviceFlushGPUDirectRDMAWrites(cudaFlushGPUDirectRDMAWritesTargetCurrentDevice,
```

Comment:
Does it work with dlopen case? Is `cudaDeviceFlushGPUDirectRDMAWrites` part of the `hmem_ops[cuda]` function table?

Reply:
You need to add this function to 

```
static struct cuda_ops cuda_ops = {
	.cudaMemcpy = cudaMemcpy,
	.cudaDeviceSynchronize = cudaDeviceSynchronize,
	.cudaFree = cudaFree,
	.cudaMalloc = cudaMalloc,
	.cudaGetErrorName = cudaGetErrorName,
	.cudaGetErrorString = cudaGetErrorString,
	.cuGetErrorName = cuGetErrorName,
	.cuGetErrorString = cuGetErrorString,
	.cuPointerGetAttribute = cuPointerGetAttribute,
	.cuPointerSetAttribute = cuPointerSetAttribute,
	.cuMemGetAddressRange = cuMemGetAddressRange,
	.cudaHostRegister = cudaHostRegister,
	.cudaHostUnregister = cudaHostUnregister,
	.cudaGetDeviceCount = cudaGetDeviceCount,
	.cudaGetDevice = cudaGetDevice,
	.cudaSetDevice = cudaSetDevice,
	.cudaIpcOpenMemHandle = cudaIpcOpenMemHandle,
	.cudaIpcGetMemHandle = cudaIpcGetMemHandle,
	.cudaIpcCloseMemHandle = cudaIpcCloseMemHandle
};
```

And initialize it as `cudaDeviceFlushGPUDirectRDMAWrites` (non-dlopen) and via dlsym (dlopen case) . See how other functions in the table are defined.

Reply:
done